scripts/dropconfig

```diff
@@ -0,0 +1,306 @@
+                          counter.get('reason', ''),
+                          counter.get('description', '')))
+        print(tabulate(table, drop_counter_config_header, tablefmt='simple', stralign='left'))
+
+    def print_device_capabilities(self):
+        """
+            Prints out the capabilities that this device has
+        """
+
+        device_caps = self.get_device_capabilities()
+
+        table = []
+        for counter, capabilities in device_caps.iteritems():
+            table.append((counter, capabilities.get('count', 'N/A')))
+        print(tabulate(table, drop_counter_capability_header, tablefmt='simple', stralign='left'))
+
+        for counter, capabilities in device_caps.iteritems():
+            supported_reasons = deserialize_reason_list(capabilities.get('reasons', ''))
+            if supported_reasons:
+                print('\n{}'.format(counter))
+                for reason in supported_reasons:
+                    print('\t{}'.format(reason))
+    
+    def create_counter(self, counter_name, alias, group, counter_type, description, reasons):
+        """
+            Creates a new counter configuration
+        """
+
+        if not counter_name:
+            raise Exception('Counter name not provided')
+        
+        if not counter_type:
+            raise Exception('Counter type not provided')
+
+        if not reasons:
+            raise Exception('No drop reasons provided')
+        
+        if self.counter_name_in_use(counter_name):
+            raise Exception('Counter name \'{}\' already in use'.format(counter_name))
+
+        # TODO: Ideally we would like to be able to check the device capabilities
+        # to see if the counter type and reasons are supported on this device. Once
+        # these features have broader platform support we can incorporate these checks.
+ 
+        for reason in reasons:
+            self.config_db.set_entry(self.config_db.serialize_key((DROP_REASON_TABLE_PREFIX, counter_name)), reason, {})
+
+        drop_counter_entry = {'type' : counter_type}
+
+        if alias:
+            drop_counter_entry['alias'] = alias
+        if group:
+            drop_counter_entry['group'] = group
+        if description:
+            drop_counter_entry['desc'] = description
+
+        self.config_db.set_entry(DEBUG_COUNTER_CONFIG_TABLE, counter_name, drop_counter_entry)
+
+    def delete_counter(self, counter_name):
+        """
+            Deletes a given counter configuration
+        """
+
+        if not counter_name:
+            raise Exception('No counter name provided')
+
+        if not self.counter_name_in_use(counter_name):
+            raise Exception('Counter \'{}\' not found'.format(counter_name))
+
+        self.config_db.set_entry(DEBUG_COUNTER_CONFIG_TABLE, counter_name, None)
+        self.config_db.delete_table(self.config_db.serialize_key((DROP_REASON_TABLE_PREFIX, counter_name)))
+
+    def add_reasons(self, counter_name, reasons):
```

Comment:
Do user need to reset the drop counter explicitly when drop_reasons are added/removed?

Reply:
Good catch! I don't believe the counter gets cleared when drop reasons are added and removed from the counter but I need to verify.

Reply:
They don't appear to be automatically cleared when drop reasons are added/removed. Do you have any concerns with this approach?

Reply:
The counters *are* cleared when drop reasons are added and removed.